src/databases/Mili/avtMiliFileFormat.C

```diff
@@ -74,4384 +53,3818 @@ extern "C" {
-        if (fpts) delete [] fpts;
-        return rv;
-    }
+    visitTimer->StopTimer(openDBTimer, "MILI: Opening database");
+}
+
+
+// ****************************************************************************
+//  Method:  avtMiliFileFormat::GetNodePositions
+//
+//  Purpose:
+//      Retrieve the node positions for the given timestep.
+//
+//  Arguments:
+//    timestep   The timestep of interest.
+//    dom        The domain of interested.
+//    meshId     The mesh id of interest. 
+//
+//  Programmer:  Alister Maguire
+//  Creation:    
+//
+//  Modifications:
+//
+// ****************************************************************************
 
+vtkPoints *
+avtMiliFileFormat::GetNodePositions(int timestep, 
+                                    int dom, 
+                                    int meshId)
+{
     //
-    // Element status' are stored in the "sand" variable 
+    // The node positions are stored in 'nodpos'.
     //
-    vtkFloatArray *sand_arr = (vtkFloatArray *) GetVar(ts, dom, "sand");
-    if (sand_arr->GetNumberOfTuples() != ncells[dom][mesh_id])
-    {
-        EXCEPTION2(UnexpectedValueException, ncells[dom][mesh_id],
-                                             sand_arr->GetNumberOfTuples());
-    }
-    float *sand_vals = (float*) sand_arr->GetVoidPointer(0);
+    char npChar[128];
+    sprintf(npChar, "nodpos");
+    char *npCharPtr = (char *)npChar;
+
+    MiliVariableMetaData *nodpos = miliMetaData[meshId]->
+        GetVarMDByShortName("nodpos", "node");
 
-    const unsigned char MESH = 'm';
-    const unsigned char FREE = 'f';
-    unsigned char *ns = new unsigned char[nnodes[dom][mesh_id]];
-    memset(ns, FREE, nnodes[dom][mesh_id]);
+    vtkPoints *vtkNodePos = NULL;
 
     //
-    // Populate nodal status array based on element status'
+    // There are datasets whose nodes remain constant throughout time. 
+    // In these cases, we will not have a "nodpos" to retrieve and
+    // will instead rely on the initial positions retrieved during
+    // ReadMesh. 
     //
-    int cell, node;
-    int num_free = nnodes[dom][mesh_id];
-    for (cell = 0; cell < ncells[dom][mesh_id]; cell++)
+    if (nodpos != NULL)
     {
-        if (sand_vals[cell] > 0.5) // element status is "good"
-        {
-            vtkIdType npts = 0, *pts = 0;
-            rv->GetCellPoints(cell, npts, pts);
-            if (npts && pts)
-            {
-                for (node = 0; node < npts; node++)
-                {
-                    int nid = pts[node];
-                    if (ns[nid] != MESH)
-                    {
-                        ns[nid] = MESH;
-                        num_free--;
-                    }
-                }
-            }
-        }
-    }
-    sand_arr->Delete();
+        int numNodes = miliMetaData[meshId]->GetNumNodes(dom);
+        int nPts     = dims * numNodes;
+        int vType    = nodpos->GetNumType();
+        int subrec   = nodpos->GetSubrecIds(dom)[0];
+        float fPts[nPts];
```

Comment:
float fPts[nPts];
And similar elsewhere in this file.
 Will not compile on Windows, as the value of nPts cannot be resolved at compile time.
Need to change to either:
float *fPts = new float[nPts]; // then don't forget to delete when finished.
or use a std::vector:
std::vector<float> fPts.
Please correct all similar code (eg lines 718, 986, 1006-8,etc)

Reply:
Ah, that's right... I always forget about that Windows conflict. Thanks for finding these! I'll get on it next week.

Reply:
I think I got all of them in my last commit... I'm going to do a second pass in a sec to make sure. I'm guessing windows is okay with ```type foo[SIZE]``` if SIZE is a macro?